### PR TITLE
provide -admin-access-log-path

### LINKED ISSUE #5231: Consul Connect - Ability to configure envoy access_log_path
#### Feature Description

The access_log_path for Envoy currently defaults to /dev/null. It'd be great if this was configurable with Consul Connect.

#### Use Case(s)

Output envoy access logs to stdout or saving them to a local file.

### PR DESCRIPTION
**Pull request by i0rek**
Fixes #5231.

### CODE REVIEW

File: command/connect/envoy/envoy.go [hunk 1]
```diff
@@ -258,13 +263,19 @@ func (c *cmd) templateArgs() (*BootstrapTplArgs, error) {
 		cluster = c.sidecarFor
 	}
 
+	adminAccessLogPath := c.adminAccessLogPath
+	if len(adminAccessLogPath) == 0 {
```
Comment (freddygv) [inline command/connect/envoy/envoy.go]: Would this be a little clearer with `if adminAccessLogPath == ""`? Since it's a string.

**Comment by i0rek** [comment]
> In the original issue one of the use cases was to output access logs to stdout, how would they do that with this flag?

Instead of running envoy with `consul connect envoy` you can pass the `-bootstrap` flag which only outputs the bootstrap configuration and run envoy on your own. This is probably what you want to do in production anyways to make sure it is being restarted and monitored. If you are using `-admin-access-log-path /dev/stdout` with `-bootstrap` you are free to look at `stdout` or pipe it somewhere else.

File: command/connect/envoy/envoy.go (hunk 1 above)
Comment (i0rek) [reply command/connect/envoy/envoy.go]: I always wonder about the same thing and I read the first answer of https://stackoverflow.com/questions/18594330/what-is-the-best-way-to-test-for-an-empty-string-in-go. If you find `== ""` better, I would be happy to change it. I don't have an opinion on this one.

**Comment by banks** [comment]
> This is probably what you want to do in production anyways to make sure it is being restarted and monitored.

Just for the record, using the exec feature of `consul connect envoy` is also perfectly valid in production provided you are supervising that and restarting on crashes etc. The `-bootstrap` option is useful when you are running Envoy in a container and want to use the official Envoy image without having to build your own hybrid with both consul and envoy binaries.

File: command/connect/envoy/bootstrap_tpl.go [hunk 2]
```diff
@@ -27,6 +27,11 @@ type BootstrapTplArgs struct {
 	// TLS is enabled.
 	AgentCAFile string
 
+	// AdminAccessLogPath The path to write the access log for the
+	// administration server. If no access log is desired specify
+	// "/dev/null".
```
Comment (banks) [inline command/connect/envoy/bootstrap_tpl.go]: Should we just make /dev/null be the default so specifying empty is also valid? This is kinda internal only but we do a bunch of default setting before the template already so this seems like a reasonable approach here too.

File: command/connect/envoy/envoy.go (hunk 1 above)
Comment (freddygv) [reply command/connect/envoy/envoy.go]: I think `== ""` would be clearer and more consistent. I did a search through the Consul repo and found that `len(x) == 0)` didn't seem to ever be used with strings. 

Thanks!

File: command/connect/envoy/bootstrap_tpl.go (hunk 2 above)
Comment (i0rek) [reply command/connect/envoy/bootstrap_tpl.go]: I made it so that it can also be left empty and then the default would be picked up.

**Comment by i0rek** [comment]
@banks I changed the comments and flag description as well as my comment above re how to run envoy. 👍

**Pull request merged**